test/integration/testpaths/testpath_migration_matrix.py

```diff
@@ -187,56 +155,59 @@ def setup_source(self, source):
             account=self.account.name,
             domainid=self.account.domainid,
         )
-
-        print("created volume with id: '{id}' and name: '{name}".format(id=vol.id, name=vol.name))
-
-        # NOTE either self.cleanup.append(vol) at this point or the more localised/explicit remove(vol) in the test method
-        # TODO improve marvin so cleanup doesn't break if an explicit remove had already be done
-
-        # make sure the volume is implemented
+        print(">>> Attach volume to the VM")
         self.vm.attach_volume(self.userapiclient, vol)
+        print(">>> Detach volume from the VM")
         self.vm.detach_volume(self.userapiclient, vol)
 
-        #put it on the origin point of this migration test
-        vol.migrate(self.userapiclient, storageid=source.id)
-
-        return vol
-
-    def remove_volume(self, volume):
-        """
+        print(">>> Starting migrations")
+        for source_name in self.source_pool_dimension:
+            for target_name in self.targets_pool_dimension:
+                if source_name != target_name:
+                    self.test_a_migration(vol, source_name, target_name)
 
-        :param volume: the id of the volume
-        :return:
-        """
-        volume.delete(self.apiclient)
+        print(">>> Delete volume")
+        vol.delete(self.apiclient)
         return
 
-    def migrate(self, volume, target):
-        """
-        as the method signature sugests
-
-        :param volume: the object to migrate
-        :param target: pool to migrate to
-        :return: None if no target pool found, the result of the migration command otherwise
-        """
-        pools = StoragePool.listForMigration(
-            self.userapiclient,
-            id=volume.id
-        )
-
-        for pool in pools:
-            if pool.name == target.name:
-                return volume.migrate(self.userapiclient, storageid=pool.id)
-        return None
-
-    def validate(self, volume, migration_result, target):
+    def test_a_migration(self, volume, source_name, target_name):
         """
+        Do a single iteration of the test sequence,
+         moving the volume to the source destination and then testing the actual migration from there to the target
 
-        :param volume:
-        :param target:
+        :param volume: the volume we want to migrate
+        :param source_name: source pool to migrate from
+        :param target_name: target pool to migrate to
         :return:
         """
-        if migration_result == None:
-            raise Exception("No migration result to verify")
-        print migration_result
-        return
\ No newline at end of file
+        # TODO validate pool results
+        source_pool = StoragePool.list(self.apiclient, name=source_name)[0]
+        target_pool = StoragePool.list(self.apiclient, name=target_name)[0]
+        voldata = volume.list(self.apiclient, id=volume.id)
+
+        # Move our volume to the source datastore
```

Comment:
having this comment indicates that we should have a method call here i.e. `move_source_volume_to_source_datastore()`
and then move the block below to a
```
    def move_source_volume_to_source_datastore(...)
```